File: src/main/java/org/jukeboxmc/block/BlockBedrock.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.ItemBedrock;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockBedrock extends Block {

    public BlockBedrock() {
        super( "minecraft:bedrock" );
    }

    @Override
    public ItemBedrock toItem() {
        return new ItemBedrock();
    }

    @Override
    public BlockType getBlockType() {
        return BlockType.BEDROCK;
    }

    @Override
    public double getHardness() {
        return -1;
    }

    @Override
    public boolean canBreakWithHand() {
        return false;
    }

    public BlockBedrock setInfiniBurn( boolean infiniBurn ) {
        return this.setState( "infiniburn_bit", infiniBurn ? (byte) 1 : (byte) 0 );
    }

    public boolean isInfiniBurn() {
        return this.stateExists( "infiniburn_bit" ) && this.getByteState( "infiniburn_bit" ) == 1;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockBedrock.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.Bedrock
import org.jukeboxmc.server.block.JukeboxBlock
import org.jukeboxmc.server.extensions.toByte

class BlockBedrock(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates),
    Bedrock {

    override fun isInfiniburn(): Boolean {
        return this.getBooleanState("infiniburn_bit")
    }

    override fun setInfiniburn(value: Boolean): BlockBedrock {
        return this.setState("infiniburn_bit", value.toByte())
    }
}